addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2336,21 +2369,51 @@ class SnippetsMenu extends Component {
      */
     getSnippetsByCategories() {
         const categories = new Map();
-        const search = this.state.search.toLowerCase();
-        const strMatches = str => !search || str.toLowerCase().includes(search);
-        for (const snippet of this.snippets.values()) {
+        let snippets = Array.from(this.snippets.values());
+        let strMatches = null;
+        let hasCustomStructureSnippet = false;
+
+        if (this.hasSnippetGroups) {
+            // We only show "categories" and "inner content" snippets in the side
+            // panel. Other snippets are displayed in the modal.
+            snippets = snippets.filter(snippet => {
+                return snippet.category.id !== "snippet_structure"
+                    && snippet.category.id !== "snippet_custom"
+                    || snippet.category.id === "snippet_custom"
+                    && ["content", "hybrid"].includes(snippet.fromCategory);
+            });
+
+            hasCustomStructureSnippet = Array.from(this.snippets.values())
+                .some(snippet => snippet.isCustom
+                    && ["structure", "hybrid"].includes(snippet.fromCategory));
```

Comment:
You could compute this before the filter (on l.2379) and use `snippets` instead of casting again the values to a new array.